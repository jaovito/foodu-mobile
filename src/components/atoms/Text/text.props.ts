import theme from '@theme';
import {ReactNode} from 'react';
import {TextProps as RNTextProps} from 'react-native/types';

export interface TextProps extends RNTextProps {
  children: ReactNode;
  fontWeight?: keyof typeof theme.fontFamily;
}

export interface TextStyleProps extends TextProps {}
