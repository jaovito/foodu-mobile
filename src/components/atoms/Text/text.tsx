import * as React from 'react';
import {Container} from './text.styles';
import {TextProps} from './text.props';

export const Text = ({children, fontWeight, ...rest}: TextProps) => (
  <Container fontWeight={fontWeight} {...rest}>
    {children}
  </Container>
);
