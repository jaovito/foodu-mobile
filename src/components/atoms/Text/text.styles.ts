import {RFValue} from 'react-native-responsive-fontsize';
import styled from 'styled-components/native';
import {TextStyleProps} from './text.props';

export const Container = styled.Text<TextStyleProps>`
  font-size: ${RFValue(16)}px;
  font-family: ${({theme, fontWeight}) =>
    theme.fontFamily[fontWeight ?? 'regular']};
  color: ${({theme}) => theme.colors.oil};
`;
