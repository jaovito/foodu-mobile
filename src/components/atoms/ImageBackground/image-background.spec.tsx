import {render} from '@test-utils';
import {ImageBackground} from '.';
import {Text} from '../Text';

describe('ImageBackground Tests', () => {
  const text = 'John doe';

  it('should render correctly', () => {
    const backgroundImage = render(
      <ImageBackground>
        <Text>{text}</Text>
      </ImageBackground>,
    );

    expect(backgroundImage).toBeDefined();
  });

  it('should be able to find the text inside the background image', async () => {
    const {findByText} = render(
      <ImageBackground>
        <Text>{text}</Text>
      </ImageBackground>,
    );

    const findText = await findByText(text);

    expect(findText).toBeTruthy();
  });

  it('should be not able to find the text that doesn`t exists in the background image', async () => {
    const {queryByText} = render(
      <ImageBackground>
        <Text>{text}</Text>
      </ImageBackground>,
    );

    const findText = queryByText('Potato');

    expect(findText).toBeNull();
  });

  it('should match with snapshot', async () => {
    const backgroundImage = render(
      <ImageBackground>
        <Text>{text}</Text>
      </ImageBackground>,
    ).toJSON;

    expect(backgroundImage).toMatchSnapshot();
  });
});
