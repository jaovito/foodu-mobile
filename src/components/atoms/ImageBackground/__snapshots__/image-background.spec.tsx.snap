// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBackground Tests should match with snapshot 1`] = `[Function]`;
